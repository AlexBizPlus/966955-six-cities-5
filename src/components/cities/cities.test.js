import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Cities from "@cities";

describe(`Render Cities component`, () => {

  const initialStore = {
    HOTELS: {
      hotels: [{
        "city": {
          "name": `Amsterdam`
        }}],
      unsorted: [{
        "city": {
          "name": `Amsterdam`
        }}],
      offer: {},
      nearby: [{}],
      update: 0,
      sort: `popular`
    },
    CITY: {
      activeCity: `Paris`,
    },
    REVIEWS: {
      reviews: [],
      reviewsCount: 0,
      review: {},
      rating: 5,
      isLoading: true,
      isError: true
    },
    MAP: {
      hover: [0, 0],
    },
    USER: {
      authorizationStatus: `AUTH`,
      login: `q@q.qq`
    }
  };

  const mockStore = configureStore([]);

  test(`Render Cities`, () => {
    const store = mockStore(initialStore);

    store.dispatch = jest.fn();

    const tree = renderer
      .create(
          <Provider store={store}>
            <Cities />
          </Provider>
      );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
