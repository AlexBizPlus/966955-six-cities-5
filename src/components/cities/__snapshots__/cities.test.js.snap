// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Cities component Render Cities 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <button
      className="locations__item-link tabs__item button_link tabs__item--active"
      onClick={[Function]}
      type="button"
      value="Paris"
    >
      <span>
        Paris
      </span>
    </button>
  </li>
  <li
    className="locations__item"
  >
    <button
      className="locations__item-link tabs__item button_link"
      onClick={[Function]}
      type="button"
      value="Cologne"
    >
      <span>
        Cologne
      </span>
    </button>
  </li>
  <li
    className="locations__item"
  >
    <button
      className="locations__item-link tabs__item button_link"
      onClick={[Function]}
      type="button"
      value="Brussels"
    >
      <span>
        Brussels
      </span>
    </button>
  </li>
  <li
    className="locations__item"
  >
    <button
      className="locations__item-link tabs__item button_link"
      onClick={[Function]}
      type="button"
      value="Amsterdam"
    >
      <span>
        Amsterdam
      </span>
    </button>
  </li>
  <li
    className="locations__item"
  >
    <button
      className="locations__item-link tabs__item button_link"
      onClick={[Function]}
      type="button"
      value="Hamburg"
    >
      <span>
        Hamburg
      </span>
    </button>
  </li>
  <li
    className="locations__item"
  >
    <button
      className="locations__item-link tabs__item button_link"
      onClick={[Function]}
      type="button"
      value="Dusseldorf"
    >
      <span>
        Dusseldorf
      </span>
    </button>
  </li>
</ul>
`;
